services/powerbi/powerbi.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/adx-automation-services/services/powerbi/kubeutils"
+	"github.com/Azure/adx-automation-services/services/powerbi/logging"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+type PowerBIService interface {
```

Comment:
What's the benefit of the two steps definition here: an interface + a struct. Are there potential different implementation?

Reply:
I am following convention from https://github.com/go-kit/kit

Reply:
Oh, good to know.